modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -266,12 +266,12 @@ object CurrencySnapshotCreator {
             Option.when(currencySnapshotAcceptanceResult.info.lastMessages.nonEmpty)(
               currencySnapshotAcceptanceResult.messages.accepted.toSortedSet
             ),
-            currencySnapshotAcceptanceResult.globalSnapshotSync.accepted.toSortedSet.some,
+            currencySnapshotAcceptanceResult.globalSnapshotSync.accepted.toSortedSet.some.filter(_.nonEmpty),
             currencySnapshotAcceptanceResult.feeTransactions,
             currencySnapshotAcceptanceResult.sharedArtifacts.some,
-            currencySnapshotAcceptanceResult.allowSpendBlock.accepted.toSortedSet.some,
-            currencySnapshotAcceptanceResult.tokenLockBlock.accepted.toSortedSet.some,
-            currencySnapshotAcceptanceResult.globalSyncView.some
+            currencySnapshotAcceptanceResult.allowSpendBlock.accepted.toSortedSet.some.filter(_.nonEmpty),
+            currencySnapshotAcceptanceResult.tokenLockBlock.accepted.toSortedSet.some.filter(_.nonEmpty),
+            currencySnapshotAcceptanceResult.globalSyncView.some.filter(_.ordinal > SnapshotOrdinal.MinValue)
```

Comment:
The other filters seem totally fine, but is there an explicit reason for this? The type itself here is already implementing NonNegLong validator? Is there a specific reason to exclude 0?